ci/docker/run.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+# Prep CI Enviroment
+if [[ "${CI}" = "true" ]]; then
+    set -exo pipefail
+fi
+
+# Receive and set variables.
+if [[ "${TRAVIS}" = "true" ]]; then
+    echo "Running build on Travis CI"
+    branch_name="dev"
+    git_repo="https://github.com/rchain/rchain"
+    docker_dst_repo="rchain/rnode:${TRAVIS_BRANCH}"
+elif [[ $1 && $2 && $3 ]]; then
+    echo "Running custom build"
+    branch_name=$1
+    git_repo=$2
+    docker_dst_repo="$3"
+else
+    echo "Invalid number of parameters."
+    echo "Example: sudo $0 <branch name> <repo url> <docker hub repo:tag>"
+    echo "Example: sudo $0 dev https://github.com/rchain/rchain myrepo/rnode:mytagname"
+    echo "You will be asked for you Docker repo user/password." 
+    exit
+fi
+
+## Prep docker image 
+# PUSHER_DOCKER_NAME="rchain-pusher-$(mktemp | awk -F. '{print $2}')"
+# Use the above vs below if you want unique build names every time. 
+# If build fails before last line with "rm -f" this can leave garbage containers running. 
+PUSHER_DOCKER_NAME="rchain-pusher-tmp"
+
+# If container exists force remove it.
+if [[ $(docker ps -aq -f name=${PUSHER_DOCKER_NAME}) ]]; then
+    docker rm -f ${PUSHER_DOCKER_NAME}
+fi
+
+if [[ "${TRAVIS}" = "true" ]]; then
+    # Start docker container with access to docker.sock so it can run view/run docker images.
+    docker run -dit -v /var/run/docker.sock:/var/run/docker.sock \
+        -e DOCKER_USERNAME="${DOCKER_USERNAME}" \
+        -e DOCKER_PASSWORD="${DOCKER_PASSWORD}" \
+        -e TRAVIS="${TRAVIS}" -e TRAVIS_BRANCH=${TRAVIS_BRANCH} \
+        --name ${PUSHER_DOCKER_NAME} ubuntu:16.04
+    # Be aware of what "-v /var/run/docker.sock:/var/run/docker.sock" is doing above.
+    # See https://jpetazzo.github.io/2015/09/03/do-not-use-docker-in-docker-for-ci/.
```

Comment:
I can't tell if this comment is telling me you're doing something bad or not. Perhaps rephrase to something like, "we don't use docker-in-docker for CI, here's why."

Reply:
I'll just remove it. It was more informational for people interested. We actually want to use docker-in-docker. Makes build easily and %100 portable. It isn't exposing host or builder to any more security risks than not running it as you still need root/sudo to run docker containers in general.

Reply:
The web page you linked to suggests that "docker-in-docker" is not the same as simply running the CI in docker. The advice he gives here is that you should not use "did" but just expose the docker socket, which looks to be what you did. Just want clarity that we don't have to worry about data corruption or the weird copy-on-write stuff he was describing in that page.